### DISCUSSION TITLE
Hybrid DSG and SSG

### DISCUSSION THREAD

**Discussion by marceloverdijk** [gatsbyjs/gatsby#34671] [Help] [answered]
Reading about DSG (Deferred Static Generation) is exactly what I would like in development mode.
There is no reason to generate all pages to avoid long build times.

However when going to prod a full SSG is desired for hosting on a CDN without having a running build server.

Is that something that could work? DSG for development and SSG for production using the same project?

**Comment by LekoArts** [comment, marked as answer]
This is already happening with "Query on Demand" since 2.30: https://www.gatsbyjs.com/docs/reference/release-notes/v2.30/#query-on-demand-and-lazy-images-generally-available
